src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -140,23 +140,9 @@ internal Conversions Conversions
         private ImmutableHashSet<SyntaxTree>? _usageOfUsingsRecordedInTrees = ImmutableHashSet<SyntaxTree>.Empty;
 
         /// <summary>
-        /// Nullable analysis data for methods, parameter default values, and attributes.
-        /// The key is a symbol for methods or parameters, and syntax for attributes.
-        /// The data is collected during testing only.
+        /// Optional data collected during testing only.
         /// </summary>
-        internal NullableData? NullableAnalysisData;
-
-        internal sealed class NullableData
-        {
-            internal readonly int MaxRecursionDepth;
-            internal readonly ConcurrentDictionary<object, NullableWalker.Data> Data;
-
-            internal NullableData(int maxRecursionDepth = -1)
-            {
-                MaxRecursionDepth = maxRecursionDepth;
-                Data = new ConcurrentDictionary<object, NullableWalker.Data>();
-            }
-        }
+        internal object? CompilationData;
```

Comment:
>CompilationData [](http://example.com/codeflow?start=25&length=15)

Consider reflecting test-only nature of this field in the name. For example, "TestOnlyCompilationData".